File: extensions/twitch/src/main/java/app/revanced/extension/twitch/adblock/IAdblockService.java
package app.revanced.extension.twitch.adblock;

import okhttp3.Request;

public interface IAdblockService {
    static boolean isVod(Request request) {
        return request.url().pathSegments().contains("vod");
    }

    static String channelName(Request request) {
        for (String pathSegment : request.url().pathSegments()) {
            if (pathSegment.endsWith(".m3u8")) {
                return pathSegment.replace(".m3u8", "");
            }
        }
        return null;
    }

    String friendlyName();

    Integer maxAttempts();

    Boolean isAvailable();

    Request rewriteHlsRequest(Request originalRequest);
}

File: integrations/java/app/revanced/twitch/adblock/IAdblockService.kt
package app.revanced.twitch.adblock

import okhttp3.Request

interface IAdblockService {
    fun friendlyName(): String
    fun maxAttempts(): Int
    fun isAvailable(): Boolean
    fun rewriteHlsRequest(originalRequest: Request): Request?

    companion object {
        fun Request.isVod() = url.pathSegments.contains("vod")
        fun Request.channelName() =
            url.pathSegments
                .firstOrNull { it.endsWith(".m3u8") }
                .run { this?.replace(".m3u8", "") }
    }
}